File: src/Colinapp.Domain/Entities/System/Role.cs
using Colinapp.Domain.Enums;

namespace Colinapp.Domain.Entities.System;

/// <summary>
/// 角色。功能权限的载体（RoleMenu 绑定菜单/按钮），并携带数据权限范围（DataScope）。
/// </summary>
public class Role : EntityBase
{
    /// <summary>角色名称</summary>
    public string Name { get; set; } = string.Empty;

    /// <summary>角色编码（唯一，如 admin、common）</summary>
    public string Code { get; set; } = string.Empty;

    /// <summary>数据权限范围</summary>
    public DataScope DataScope { get; set; } = DataScope.All;

    /// <summary>显示排序</summary>
    public int OrderNum { get; set; }

    /// <summary>状态：true=启用，false=停用</summary>
    public bool Enabled { get; set; } = true;

    /// <summary>备注</summary>
    public string? Remark { get; set; }
}

File: web/src/api/role.ts
import { http } from '@/utils/request'
import type { PagedQuery, PagedResult } from './types'

export const enum DataScope {
  All = 1,
  Custom = 2,
  Dept = 3,
  DeptAndChild = 4,
  Self = 5,
}

export interface RoleItem {
  id: number
  name: string
  code: string
  dataScope: DataScope
  orderNum: number
  enabled: boolean
  remark?: string
  createdTime: string
}

export interface RoleDetail extends RoleItem {
  menuIds: number[]
  deptIds: number[]
}

export interface RoleSaveDto {
  name: string
  code: string
  dataScope: DataScope
  orderNum: number
  enabled: boolean
  remark?: string
  menuIds: number[]
  deptIds: number[]
}

export const getRoles = (params: PagedQuery) => http.get<PagedResult<RoleItem>>('/role', params)
export const getRoleOptions = () => http.get<RoleItem[]>('/role/options')
export const getRole = (id: number) => http.get<RoleDetail>(`/role/${id}`)
export const createRole = (data: RoleSaveDto) => http.post('/role', data)
export const updateRole = (id: number, data: RoleSaveDto) => http.put(`/role/${id}`, data)
export const deleteRole = (id: number) => http.delete(`/role/${id}`)
